jxline-core/src/main/java/cloud/xline/jxline/Txn.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016-2023 The jetcd authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloud.xline.jxline;
+
+import cloud.xline.jxline.kv.TxnResponse;
+import io.etcd.jetcd.op.Cmp;
+import io.etcd.jetcd.op.Op;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Txn is the interface that wraps mini-transactions.
+ *
+ * <h2>Usage examples</h2>
+ *
+ * <pre>{@code
+ * txn.If(
+ *     new Cmp(KEY, Cmp.Op.GREATER, CmpTarget.value(VALUE)),
+ *     new Cmp(KEY, cmp.Op.EQUAL, CmpTarget.version(2))).Then(
+ *         Op.put(KEY2, VALUE2, PutOption.DEFAULT),
+ *         Op.put(KEY3, VALUE3, PutOption.DEFAULT))
+ *     .Else(
+ *         Op.put(KEY4, VALUE4, PutOption.DEFAULT),
+ *         Op.put(KEY4, VALUE4, PutOption.DEFAULT))
+ *     .commit();
+ * }</pre>
+ *
+ * <p>Txn also supports If, Then, and Else chaining. e.g.
+ *
+ * <pre>{@code
+ * txn.If(
+ *     new Cmp(KEY, Cmp.Op.GREATER, CmpTarget.value(VALUE))).If(
+ *         new Cmp(KEY, cmp.Op.EQUAL, CmpTarget.version(VERSION)))
+ *     .Then(
+ *         Op.put(KEY2, VALUE2, PutOption.DEFAULT))
+ *     .Then(
+ *         Op.put(KEY3, VALUE3, PutOption.DEFAULT))
+ *     .Else(
+ *         Op.put(KEY4, VALUE4, PutOption.DEFAULT))
+ *     .Else(
+ *         Op.put(KEY4, VALUE4, PutOption.DEFAULT))
+ *     .commit();
+ * }</pre>
+ */
+public interface Txn {
+    // CHECKSTYLE:OFF
+    /**
+     * takes a list of comparison. If all comparisons passed in succeed, the operations passed into
+     * Then() will be executed. Or the operations passed into Else() will be executed.
+     *
+     * @param cmps the comparisons
+     * @return this object
+     */
+    Txn If(Cmp... cmps);
```

Comment:
function name starts with lowercase?

Reply:
The meaning here should be to distinguish it from `commit`.
I think this is acceptable.

We also need to provide an API same as jetcd.
https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/Txn.java

Reply:
The naming is totally different, will we mix `if` with `commit`? I don't think uppercase captain is necessary.

Reply:
I don't think this is a good idea. When migrating jetcd, we also need to consider changing the "If" in txn requests to "if", which can be confusing.
Please focus on the `KVImpl.java`, `ProtocolClientImpl.java`, `ClientConnectionManager.java` and suggest areas for improvement, as they will be heavily used later on.